README.md

```diff
@@ -2,41 +2,82 @@
-for testing, playing, manipulating your code so that you understand it better
-and solve problems with it.
+IRB stands for "Interactive Ruby." It's a like a room especially built for
+having conversations with Ruby. Programs like IRB are known as "REPLs." IRB is
+run from within your computer's terminal.
 
-## How do you use it?
+IRB is _not_ a file where you save your work. Any coding you do in the IRB
+console in your terminal will not get saved anywhere. It only exists
+temporarily. IRB is for testing, playing, manipulating your code so that you
+understand it better and solve problems with it.
 
-IRB allows you to execute ruby in the terminal and you're going to get
-comfortable using it to test and better understand your Ruby code. To access
-IRB, just type `irb` in the terminal. IRB allows you to do anything you can do
-in a Ruby file. For instance, you can do math, get the time by typing
-`Time.now`, or print text to the screen.
+## Identify REPLs
 
-## Instructions
+REPL stands for read–eval–print loop.
+
+If you think about it, that's what you do when you're being the listener in a
+conversation.
+
+You:
+
+1. "read" in the other person's expression ("I'm curious about learning programming!")
+2. "evaluate" what the expression means ("My friend is bored at their job? My friend wants to make more money?)
+3. "print" your reponse ("Oh great!" "I love programming, you're going to love it too!").
+
+IRB, as a REPL, gives you a place where:
+
+1. _You_ type in a Ruby expression (which IRB "reads")
+2. _IRB_ "evaluates" the expression based on the rules of Ruby
+3. _IRB_ "prints" a response or a _return value_
+
+## Explain why REPLs are useful
+
+Sometimes we want to see how code works. Instead of having to build a brand new
+program (in a file), load up all the libraries, blah, blah, blah, IRB is a
+quick "laboratory" where we can see how something works.
+
+Think about a car: it's a complex machine with many interacting parts. But if
+your car won't start you might "isolate" the battery and hook it up to a
+battery tester. If the battery tester shows the battery is good, then you know
+that the problem is somewhere else; if the battery is dead, then you know right
+what to do.
+
+That process is known as _debugging_ when applied to computers or
+_troubleshooting_ when applied to pretty much everything else.
+
+> Randy counts four men in addition to Amy and the pilot...  All of them are
+> fiddling around with engines or diving gear in a way Randy recognizes,
+> through many cultural and technological barriers, as debugging.
+
+From Neal Stephenson's _Cryptonomicon_
+
+## Access and Exit IRB Via Your Terminal
+
+To access IRB, just type `irb` in the terminal. Let's start by typing a few
+_expressions_ into IRB and see the read-evaluate-print-loop for ourselves.
+
+### Instructions
 
 1. Open up your terminal. (If you do not have a terminal on your computer you
    can use [repl.it][repl])
```

Comment:
since we've got a terminal on learn.co, should we encourage use of IRB in there?